operations/research-and-development/product/development-process/secure-software-development.md

```diff
@@ -0,0 +1,85 @@
+
+## Code reviews
+
+### Manual code reviews
+
+The product security team performs reviews on new features and plugins when necessary. Here's some of the things included in the checks:
+
+* Manual diff review
+* Static application security testing (SAST) using multiple tools
+* Software composition analysis (SCA)
+* Dynamic application security testing (DAST)
+
+Detailed review checklists are documented in the [security team shared drive](https://drive.google.com/drive/folders/1s3paBN6sfGRP-NHNuoekDHY_nrtRVagt), accessible to team members only.
+
+### Automated code reviews
+
+The key Mattermost repositories have automated code security analysis tools set up. If this is not the case, please reach out to the Product Security Team for assistance in setting this up. Particular automated tools in each repository depend on the technologies used. In practice, we use [CodeQL](https://codeql.github.com/) as a Github Action to automatically analyze all pull requests. Additionally, the use of the [OSSF Security Scorecards GitHub Action](https://github.com/ossf/scorecard) is strongly encouraged. This document addresses many of the best practices checked by the Scorecards action.
+
+## Dependency pinning
+
+Dependency pinning hardens the software development workflow against [software supply chain attacks](https://arstechnica.com/information-technology/2019/08/the-year-long-rash-of-supply-chain-attacks-against-open-source-is-getting-worse/). In this context, the specific threat being mitigated is upstream dependencies being taken over by attackers and replaced with malicious content. The risk commonly involves referring to dependencies without explicitly specifying a computed hash value for the dependency. In case of a dependency takeover, pulling the _latest_ version of it realizes the risk. The risk probability is considered low for supply chain attacks, but the effect can be pretty bad.
+
+Practically, defining build-time dependencies with an explicit hash is the preferred solution. When specifying hashes to existing dependencies that did not use hashed references before, picking the hash of the currently used dependency is a good starting point as it is often not practical to do a full audit of all dependencies. Picking a hash and sticking to it doesn't ensure that said hash represents a safe version, it only ensures that a possibly malicious later version is not introduced in the build process afterwards. Some technologies support automatic dependency pinning so that the developer doesn't need to explicitly find and set dependency hashes, for example Go and npm, as described below.
+
+### Npm dependencies
+
+With npm, replace `npm install` with `npm ci` everywhere in the build/CI pipeline. This ensures that only packages matching `package-lock.json` are installed, see [the documentation on `npm ci`](https://docs.npmjs.com/cli/v8/commands/npm-ci) for more details and a reference to the npm documentation.
```

Comment:
```suggestion
With npm, always use `npm ci` instead of `npm install` in build scripts and automated pipelines. This ensures that only packages matching the existing `package-lock.json` file are installed. See [the documentation on `npm ci`](https://docs.npmjs.com/cli/v8/commands/npm-ci) for more details.
```

Minor wording suggestions + remove the superfluous ` and a reference to the npm documentation`.